### PR TITLE
soc: arm: stm32f4: Add STM32F423 support

### PR DESCRIPTION
**Pull request by rosterloh**
Signed-off-by: Richard Osterloh <richard.osterloh@gmail.com>

### CODE REVIEW

File: dts/arm/st/f4/stm32f423.dtsi [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/ {
```
Comment (FRASTM) [inline dts/arm/st/f4/stm32f423.dtsi:1]: should this dtsi include header and the `#include <st/f4/stm32f413.dtsi>`

Comment (FRASTM) [reply dts/arm/st/f4/stm32f423.dtsi:1]: Note that stm32f415 and stm32f417 already include the `cryp: cryp@50060000` node

Comment (rosterloh) [reply dts/arm/st/f4/stm32f423.dtsi:1]: @FRASTM the stm32f423 is exactly the same as the stm32f413 but has the AES module so I thought it made sense to just include everything from stm32f413. I'm happy to change based on your recommendations. stm32f417 includes crypto acceleration and as far as I understand this chip just has aes accelerator but it seems like the dts includes the same common file.

**Review by erwango — CHANGES_REQUESTED** [review]
The inclusion scheme is not correct.

Soc variant f423.dtsi should:
- include f413.dtsi
- define the additional nodes (aes in this case)
Then packages variants (Xh/g.dtsi files) should:
- include soc variant
- define the RAM/Flash sizes 

See other series for similar pattern.

**Comment by rosterloh** [comment]
@erwango is this better?

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**